server/src/org/apache/cloudstack/network/topology/AdvancedNetworkVisitor.java

```diff
@@ -61,13 +61,15 @@
     @Override
     public boolean visit(final UserdataPwdRules userdata) throws ResourceUnavailableException {
         final VirtualRouter router = userdata.getRouter();
-
         final Commands commands = new Commands(Command.OnError.Stop);
         final VirtualMachineProfile profile = userdata.getProfile();
         final NicVO nicVo = userdata.getNicVo();
         final UserVmVO userVM = userdata.getUserVM();
 
-        _commandSetupHelper.createPasswordCommand(router, profile, nicVo, commands);
+        if (router.getRedundantState() == VirtualRouter.RedundantState.MASTER) {
```

Comment:
@ustcweizhou 
I am also not sure about the VPC behavior, so for now i will not alter the behavior for vpc networks.  I have added the changes to ignore non redundant vrs and vpc vrs.